gef.py

```diff
@@ -7297,7 +7297,10 @@ def check_thread_ids(tids):
     @staticmethod
     def tcachebin(tcache_base, i):
         """Return the head chunk in tcache[i] and the number of chunks in the bin."""
-        assert i <  GlibcHeapTcachebinsCommand.TCACHE_MAX_BINS, "index should be less then TCACHE_MAX_BINS"
+        if i >= GlibcHeapTcachebinsCommand.TCACHE_MAX_BINS:
+            err("Incorrect index value, {} not in [0, {}[".format(i, GlibcHeapTcachebinsCommand.TCACHE_MAX_BINS))
```

Comment:
Wrong `[` at the end.

Also, don't we want this to raise an Exception so that its caller knows not to continue?

Reply:
> Wrong `[` at the end.

It's an exclusive interval so no that's correct.

> Also, don't we want this to raise an Exception so that its caller knows not to continue?

No, I thought about it but if the index is wrong it's probably because the heap is corrupt. It's not an unexpected behavior of GEF.

Reply:
Ah, then shouldn't it be `[0, n)`?

Reply:
[It's the same](https://en.wikipedia.org/wiki/ISO_31-11)

Reply:
Heh! TIL